lib/MusicBrainz/Server/Data/Artist.pm

```diff
@@ -465,12 +465,37 @@ sub _hash_to_row
 }
 
 sub load_related_info {
-    my ($self, @artists) = @_;
+    my ($self, $artists_ref, $user_lang) = @_;
 
     my $c = $self->c;
+    my @artists = @{$artists_ref};
     $c->model('ArtistType')->load(@artists);
     $c->model('Gender')->load(@artists);
     $c->model('Area')->load(@artists);
+
+    if (defined $user_lang) {
+        my $short_user_lang = substr($user_lang, 0, 2);
+        my $aliases = $c->model('Artist')->alias->find_by_entity_ids(
+            map { $_->id } @artists,
+        );
+        for my $artist (@artists) {
+            my $best_alias;
+            foreach my $alias (@{$aliases->{$artist->id}}) {
+                next if !defined $alias->locale || !$alias->primary_for_locale;
+                # If we find an exact match for the user's language, use it.
+                if ($alias->locale eq $user_lang) {
+                    $best_alias = $alias;
+                    last;
+                }
+                # Otherwise, favor more-generic aliases (e.g. "en" over "en_US").
```

Comment:
`_with_primary_alias` in lib/MusicBrainz/Server/WebService/JSONSerializer.pm appears to also fall back to `en` and `en_.*`. This seems like it's probably desirable behavior for e.g. a Spanish user viewing a Japanese artist who only has an English alias, but it seems incorrect in many other cases.

This is something that's always bugged me a bit about aliases: there's no way to mark an alias as primary for all Latin-script languages, and there's also no locale associated with an entity's name, only with its aliases.

Let me know if there's something better I could be doing here. I think that what I'd really like would be to fall back to an alias using the same script as the user's language if the entity name isn't already using that script, but I'm not sure if that's possible.

Reply:
> I think that what I'd really like would be to fall back to an alias using the same script as the user's language

I'm not sure if we have any mapping between language (or territory) and likely script (is this available in the CLDR?), but perhaps @yvanzo has an idea.

I do wonder if `_with_primary_alias` could be modified to make use of the proposed utility function, so that it can benefit from any future improvements in this area too.

Reply:
If it's okay with you, I'd prefer to avoid changing `_with_primary_alias` in this PR to reduce the (already high-seeming to me) odds that I break existing functionality. :-) Is adding a TODO okay for now?